### PR TITLE
ref(replays): Parameterize the transcoder's function

### PR DESCRIPTION
**Pull request by jan-auer**
An internal refactor to unblock #1731.

#skip-changelog

### CODE REVIEW

**Review by olksdr — APPROVED** [review]
lgtm

File: relay-replays/src/transform.rs [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+/// struct Identity;
+///
+/// let json = "42";
+/// let json_deserializer = &mut serde_json::Deserializer::from_str(&json);
+/// let deserializer = TransformingDeserializer::new(json_deserializer, Identity);
+///
+/// let number = deserializer.deserialize_u32(deserializer).unwrap();
+/// assert_eq!(number, 42);
+/// ```
+pub struct TransformingDeserializer<'a, D, T>(D, Mut<'a, T>);
```
Comment (jjbayer) [inline relay-replays/src/transform.rs]: nit: change to named fields for readability?

File: relay-replays/src/transform.rs [hunk 2]
```diff
@@ -0,0 +1,715 @@
...
+    fn deserialize_ignored_any<V>(mut self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        self.0
+            .deserialize_ignored_any(Visitor(visitor, self.1.as_mut()))
+    }
+}
+
+struct Visitor<'a, V, T>(V, &'a mut T);
```
Comment (jjbayer) [inline relay-replays/src/transform.rs:447]: nit: Rename to `TransformingVisitor` to be consistent with `TransformingSerializer`? Or drop the prefix on both.

Comment (jjbayer) [inline relay-replays/src/transform.rs:447]: nit: Name fields for readability?

**Review by jjbayer — APPROVED** [review]

File: relay-replays/src/transform.rs (hunk 1 above)
Comment (jan-auer) [reply relay-replays/src/transform.rs]: Tried it out, but I have to admit that this leads to a bit too much line wrapping for my taste. If it's OK, I would leave it this way.

File: relay-replays/src/transform.rs (hunk 2 above)
Comment (jan-auer) [reply relay-replays/src/transform.rs:447]: Going to rename the Deserializer instead.

**Pull request merged**